aws/resource_aws_glue_dev_endpoint_test.go

```diff
@@ -0,0 +1,945 @@
+	publicKey4                    = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQD3F6tyPEFEzV0LX3X8BsXdMsQz1x2cEikKDEY0aIj41qgxMCP/iteneqXSIFZBp5vizPvaoIR3Um9xK7PGoW8giupGn+EPuxIA4cDM4vzOqOkiMPhz5XK0whEjkVzTo4+S0puvDZuwIsdiW9mxhJc7tgBNL0cYlWSYVkz4G/fslNfRPW5mYAM49f4fhtxPb5ok4Q2Lg9dPKVHO/Bgeu5woMc7RY0p1ej6D4CKFE6lymSDJpW0YHX/wqE9+cfEauh7xZcG0q9t2ta6F6fmX0agvpFyZo8aFbXeUBr7osSCJNgvavWbM/06niWrOvYX2xwWdhXmXSrbX8ZbabVohBK41 foo4@bar.com"
+)
+
+func init() {
+	resource.AddTestSweepers("aws_glue_dev_endpoint", &resource.Sweeper{
+		Name: "aws_glue_dev_endpoint",
+		F:    testSweepGlueDevEndpoint,
+	})
+}
+
+func testSweepGlueDevEndpoint(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting client: %s", err)
+	}
+	conn := client.(*AWSClient).glueconn
+
+	input := &glue.GetDevEndpointsInput{}
+	err = conn.GetDevEndpointsPages(input, func(page *glue.GetDevEndpointsOutput, lastPage bool) bool {
+		if len(page.DevEndpoints) == 0 {
+			log.Printf("[INFO] No Glue Dev Endpoints to sweep")
+			return false
+		}
+		for _, endpoint := range page.DevEndpoints {
+			name := aws.StringValue(endpoint.EndpointName)
+			if !strings.HasPrefix(name, GlueDevEndpointResourcePrefix) {
+				log.Printf("[INFO] Skipping Glue Dev Endpoint: %s", name)
+				continue
+			}
+
+			log.Printf("[INFO] Deleting Glue Dev Endpoint: %s", name)
+			_, err := conn.DeleteDevEndpoint(&glue.DeleteDevEndpointInput{
+				EndpointName: aws.String(name),
+			})
+			if err != nil {
+				log.Printf("[ERROR] Failed to delete Glue Dev Endpoint %s: %s", name, err)
+			}
+		}
+		return !lastPage
+	})
+	if err != nil {
+		if testSweepSkipSweepError(err) {
+			log.Printf("[WARN] Skipping Glue Dev Endpoint sweep for %s: %s", region, err)
+			return nil
+		}
+		return fmt.Errorf("error retrieving Glue Dev Endpoint: %s", err)
+	}
+
+	return nil
+}
+
+func TestAccGlueDevEndpoint_Basic(t *testing.T) {
+	var endpoint glue.DevEndpoint
+
+	rName := acctest.RandomWithPrefix(GlueDevEndpointResourcePrefix)
+	resourceName := "aws_glue_dev_endpoint.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSGlueDevEndpointDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGlueDevEndpointConfig_Basic(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSGlueDevEndpointExists(resourceName, &endpoint),
+
+					testAccCheckResourceAttrRegionalARN(resourceName, "arn", "glue", fmt.Sprintf("devEndpoint/%s", rName)),
+					resource.TestCheckResourceAttr(resourceName, "name", rName),
+					testAccMatchResourceAttrGlobalARN(resourceName, "role_arn", "iam", regexp.MustCompile(`role/AWSGlueServiceRole-tf-acc-test-[0-9]+$`)),
```

Comment:
lets swap this for `					resource.TestCheckResourceAttrPair(resourceName, "role_arn", roleResourceName, "role_arn"),
`